src/main/java/org/broadinstitute/hellbender/metrics/InsertSizeMetricsCollector.java

```diff
@@ -1,50 +1,66 @@
 package org.broadinstitute.hellbender.metrics;
 
-import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMFileHeader;
 import htsjdk.samtools.SAMRecord;
 import htsjdk.samtools.SamPairUtil;
 import htsjdk.samtools.metrics.MetricsFile;
 import htsjdk.samtools.reference.ReferenceSequence;
-import htsjdk.samtools.util.Histogram;
-import org.broadinstitute.hellbender.metrics.MetricAccumulationLevel;
-import org.broadinstitute.hellbender.metrics.MultiLevelCollector;
-import org.broadinstitute.hellbender.metrics.PerUnitMetricCollector;
-
+import htsjdk.samtools.util.IOUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.AuthHolder;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.filters.WellformedReadFilter;
+import org.broadinstitute.hellbender.utils.R.RScriptExecutor;
+import org.broadinstitute.hellbender.utils.R.RScriptExecutorException;
+import org.broadinstitute.hellbender.utils.io.Resource;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
 import java.io.Serializable;
-import java.util.*;
 
 /**
- * Collects InsertSizeMetrics on the specified accumulationLevels using
+ * Collects InsertSizeMetrics on the specified accumulationLevels
  */
-public final class InsertSizeMetricsCollector
-        extends MultiLevelCollector<InsertSizeMetrics, Integer, InsertSizeMetricsCollectorArgs>
-        implements Serializable {
-
+public class InsertSizeMetricsCollector
```

Comment:
why not final anymore? Is it designed to be subclassed? there are no subclasses now and let's assume the will be none